### PR TITLE
Perception: Real-time ego pose adjustment for the change of pitch angle between camera and ground plane

### PR DESCRIPTION
**Pull request by eugenegx**
Per-frame Pitch angle real-time calibration based on the assumption that objects should be on the ground

### CODE REVIEW

File: modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc [hunk 1]
```diff
@@ -78,5 +97,70 @@ Eigen::Matrix<float, 3, 1> FlatCameraTransformer::MakeUnit(
...
+bool FlatCameraTransformer::HaveHighConfidence(
+  std::shared_ptr<VisualObject> obj_ptr) {
+    if (obj_ptr->trunc_width > 0.25f) return false;
+    if (obj_ptr->trunc_height > 0.25f) return false;
+    if (obj_ptr->distance > 50.0) return false;
+    if (obj_ptr->distance < 5.0) return false;
+
+    double azimuth = std::atan2(obj_ptr->center[1], obj_ptr->center[0])
+     * 180.0 / M_PI;
+    if (!(-30.0 < azimuth && azimuth < 30.0)) return false;
```
Comment (Capri2014) [inline modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc]: Try convert it into conf or flag if possible.

File: modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc [hunk 2]
```diff
@@ -78,5 +97,70 @@ Eigen::Matrix<float, 3, 1> FlatCameraTransformer::MakeUnit(
...
+      return;
+    }
+
+    float min_diff = std::numeric_limits<float>::max();
+    float best_pitch_adjustment = 0.0f;
+    Eigen::Matrix<float, 4, 4> best_camera2car = camera2car_;
+
+    Eigen::Matrix<float, 4, 4> rot;
+    rot.setIdentity();
+    for (float p = -3.0; p < 3.0; p += 0.2f) {
```
Comment (Capri2014) [inline modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc]: @eugenegx Are you adjusting along the yawl angle (of the vehicle) instead? between -3.0 and 3.0 with 0.2 degree of increment?

Comment (eugenegx) [reply modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc]: This is pitch :)

File: modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc (hunk 1 above)
Comment (eugenegx) [reply modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc]: Next fix

File: modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc (hunk 2 above)
Comment (Capri2014) [reply modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc]: Sorry my bad, camera2car_ already converted into FLU

**Comment by Capri2014** [comment]
LGTM

File: modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc [hunk 3]
```diff
@@ -25,14 +25,17 @@ bool FlatCameraTransformer::Transform(
     std::vector<std::shared_ptr<VisualObject>> *objects) {
   if (!objects) return false;
 
+  // Center in Camera Space and Derived Center in Ego Car Space
+  std::vector<std::pair<Eigen::Vector3f, Eigen::Vector3f>> centers;
```
Comment (Galxy027) [inline modules/perception/obstacle/camera/transformer/flat_camera_transformer.cc]: I still don't understand the meaning of the center. Can you explain it in detail?
Thank you!

**Pull request merged**